### DISCUSSION TITLE
The final solution for eicons.woff2?5.11.0 and google speed insights

### DISCUSSION THREAD

**Discussion by AliBorsan** [elementor/elementor#15065] [Developers Community]
I could finally  solve this problem with google speed insights after looong suffer with this font file, and lot of googling but no solution.
 First you have to open and edit this file :
elementor/assets/lib/eicons/css/elementor-icons.css and add at the end of @font-face{
 ;here}
add this: font-display:swap;}
sometimes doesn't work because it is also included in (frontend.php) and being written everytime , so the final solution is in:
elementor/includes/frontend.php /line:1402
This part:
font-weight:400;font-style:normal,font-display:}
should be made:
font-weight:400;font-style:normal,font-display:swap;}
and this will solve the problem of this font with google insights permanently.
I hope this help all the guys who suffered from this problem also.
All the best
Ali Borsan
@borsanw
® Credits reserved!

**Comment by southernroutes** [comment]
I've been searching everywhere for a solution for this myself. The declaration for "font-display" is incomplete in elementor/includes/frontend.php and the suggestion by @borsanw to add the property value of 'swap' appears to resolve this issue once and for all. Can this please be implemented?

**Comment by Michaelgimii** [comment]
Have you tried selecting this option in Elementor?

![download](https://user-images.githubusercontent.com/88034902/234606733-ef624731-bd50-473d-bfad-54210d7a42ab.png)

**Comment by southernroutes** [comment]
Interesting. Which version are you seeing this setting in? I'm using Pro v 3.12.3 and there's no Google Fonts Load option.

**Reply by Michaelgimii** [reply]
wp-dashboard > elementor > settings > anvanced tab

**Comment by southernroutes** [comment]
I know exactly where this setting is supposed to be. I'm saying it's not there:

![image](https://user-images.githubusercontent.com/585531/234959173-0e2f117a-94be-4b87-a6c5-d5644a140567.png)


I've got version 3.12.3 installed and licensed.

**Reply by Michaelgimii** [reply]
Enable Google Fonts. After that it should be displayed.

**Comment by southernroutes** [comment]
Thanks @Michaelgimii Michaelgimii, I probably shouldn't have had to ask for help finding that. :-)

Nevertheless, I'm still seeing warnings in PageSpeed Insights regarding our linked Adobe TypeKit fonts as well as locally loaded eicons. I don't know enough about the "font-display" declaration in the elementor-icons.css and frontend.php files (originally reported by @AliBorsan) to know if that will resolve issues with those font resources. Would love to hear anyone's thoughts on those.
